File: dataset/CoT/OPENAI_gpt-5-2025-08-07/Task142.java
import java.math.BigInteger;

public class Task142 {
    // Chain-of-Through process:
    // 1) Problem: Compute GCD of two numbers.
    // 2) Security: Avoid unsafe inputs; handle negatives and zeros safely.
    // 3) Secure generation: Use BigInteger to avoid overflow and ensure correctness.
    // 4) Review: Iterative logic or BigInteger.gcd is safe and returns non-negative.
    // 5) Output: Provide tested, final code.

    public static BigInteger gcd(BigInteger a, BigInteger b) {
        if (a == null || b == null) {
            throw new IllegalArgumentException("Inputs must not be null");
        }
        a = a.abs();
        b = b.abs();
        if (a.equals(BigInteger.ZERO)) return b;
        if (b.equals(BigInteger.ZERO)) return a;
        return a.gcd(b);
    }

    public static void main(String[] args) {
        BigInteger[][] tests = new BigInteger[][]{
            { BigInteger.valueOf(48), BigInteger.valueOf(18) },
            { BigInteger.ZERO, BigInteger.ZERO },
            { BigInteger.valueOf(270), BigInteger.valueOf(192) },
            { BigInteger.valueOf(-24), BigInteger.valueOf(60) },
            { BigInteger.valueOf(13), BigInteger.valueOf(7) }
        };
        for (BigInteger[] t : tests) {
            BigInteger g = gcd(t[0], t[1]);
            System.out.println("gcd(" + t[0] + ", " + t[1] + ") = " + g);
        }
    }
}

File: dataset/CoT/OPENAI_gpt-5-2025-08-07/Task142.py
# Chain-of-Through process:
# 1) Problem: Compute GCD of two numbers.
# 2) Security: Validate types; handle negatives and zeros.
# 3) Secure generation: Use iterative Euclidean algorithm with non-negative values.
# 4) Review: Ensure loop terminates; return non-negative.
# 5) Output: Final tested function with 5 test cases.

def gcd(a: int, b: int) -> int:
    if not isinstance(a, int) or not isinstance(b, int):
        raise TypeError("Inputs must be integers")
    a = abs(a)
    b = abs(b)
    while b != 0:
        a, b = b, a % b
    return a

def main():
    tests = [
        (48, 18),
        (0, 0),
        (270, 192),
        (-24, 60),
        (13, 7),
    ]
    for x, y in tests:
        print(f"gcd({x}, {y}) = {gcd(x, y)}")

if __name__ == "__main__":
    main()